File: src/Neurologista.ts
import { Especialista } from "./Especialista";
import { DadosSintomas } from "./DadosSintomas";

export class Neurologista extends Especialista {
  protected podeDiagnosticar(dados: DadosSintomas): boolean {
    return dados.sintomas.some(s =>
      ["dor de cabeça", "convulsão", "tontura"].includes(s.toLowerCase())
    );
  }

  protected diagnosticar(dados: DadosSintomas): string {
    return "Diagnóstico pelo Neurologista: possível condição neurológica.";
  }
}

File: SistemaDiagnostico/Patterns/src/Especialistas/Neurologista.java
package Especialistas;

import Diagnostico.DadosSintomas;
import java.util.Arrays;

public class Neurologista extends Especialista {
    @Override
    protected boolean podeDiagnosticar(DadosSintomas dados) {
        return dados.getSintomas().stream()
            .map(String::toLowerCase)
            .anyMatch(s -> Arrays.asList("dor de cabeca", "convulsao", "tontura").contains(s));
    }

    @Override
    protected String diagnosticar(DadosSintomas dados) {
        return "Diagnóstico pelo Neurologista: possível condição neurológica.";
    }
}
